docs/adr/0002-use-local-datastore.md

```diff
@@ -0,0 +1,72 @@
+
+We are about to build another [Publishing API] client and we should now use the learnings we have to ensure that not using a local datastore is the best decision.
+
+### Option 1: No local datastore
+In this option we continue with the platform architecture goals.
+
+#### Pros
++ Single source of truth - all data is in the Publishing API
++ Multiple apps can edit the same content
++ Single implementation of scheduling
++ Single abstraction layer
+
+#### Cons
+- Effectively building an Active Resource layer (object-relational mapper for REST)
+- Harder to resolve performance problems
+- Current workflow is opinionated and may not meet needs of content
+- Testing against an API is hard, it requires a lot of stubbing or running a version of the API as part of the test suite
+
+### Option 2: Use a local datastore
+#### Pros
++ It’s standard Rails so quick to build and easy to understand
++ Easy to test, testing with a local datastore is very straightforward
++ Easy to iterate workflow, changing the API workflow is much harder
++ Only final state of content can be put into Publishing API - some formats will need interim data which isn’t needed for publishing
++ No new features needed for meta data - notes, and other publisher info
+
+#### Cons
+- Syncing between local and Publishing API
+- Local data is hard to query by other applications
+- Combination of local and Publishing API workflows
+- Hard to feedback to use on asynchronous API updates
+- Content-Schema may not match Rails validations causing unhelpful error messages
+
+In addition to the above there are a few pragmatic problems which have influenced our decision.
+
+We want to store content prior to it being available as a draft on Publishing API. Doing this with no local datastore would have the complexity of a holding place for this data.
+
+We want to model the full edit history of documents, which Publishing API doesn't support.
+
+We want to support migrating of Whitehall documents with their histories - which is arguably simpler then moving to another publishing app than having to resume the Publishing API migration work
+
+We want to support features the Publishing API doesn't have such as reviews, scheduling and storing additional user level meta data.
+
+All of these can be done with a commitment to developing the Publishing API but it would much slower, would mean resuming the migration programme and potentially affecting every client of the Publishing API. It also isn't clear that Publishing API should have all of this functionality.
+
+To achieve the aims of rapidly building a new publishing application to be ready for beta we're able to do this faster and more effectively by modelling these concepts in the local database.
+
+## Decision
+We will use a local datastore which syncs to the Publishing API.
+
+## Status
+Accepted
+
+## Consequences
```

Comment:
I agree with this decision. Given that, we also need to review and update the Architecture Goals document to be consistent with it.

Reply:
@timblair I've added this as a consequence in 757f1068092548ea77cf702c4f6cd4abb08a56ab